docker/runtime/java/handler/src/main/java/io/kubeless/Handler.java

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright (c) 2016-2017 Bitnami
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package io.kubeless;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.io.InputStreamReader;
+import java.io.BufferedReader;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+import com.sun.net.httpserver.Headers;
+
+import io.prometheus.client.Counter;
+import io.prometheus.client.Histogram;
+import java.lang.reflect.InvocationTargetException;
+import io.kubeless.Event;
+import io.kubeless.Context;
+
+public class Handler {
+    public static void main(String[] args) {
+        String funcPort = System.getenv("FUNC_PORT");
+        if(funcPort == null || funcPort.isEmpty()) {
+            funcPort = "8080";
+        }
+        int port = Integer.parseInt(funcPort);
+        try {
+            HttpServer server = HttpServer.create(new InetSocketAddress(port), 0);
+            server.createContext("/", new FunctionHandler());
+            server.createContext("/healthz", new HealthHandler());
+            server.setExecutor(null);
+            server.start();
+        } catch (java.io.IOException e) {
+            System.out.println("Failed to starting listener.");
+        }
+    }
+
+    static class FunctionHandler implements HttpHandler {
+        String className = System.getenv("MOD_NAME");
+        String methodName = System.getenv("FUNC_HANDLER");
+        String timeout = System.getenv("FUNC_TIMEOUT");
+        String runtime = System.getenv("FUNC_RUNTIME");
+        String memoryLimit = System.getenv("FUNC_MEMORY_LIMIT");
+        static final Counter requests = Counter.build().name("function_calls_total").help("Total function calls.").register();
+        static final Counter failures = Counter.build().name("function_failures_total").help("Total function call failuress.").register();
+        static final Histogram requestLatency = Histogram.build().name("function_duration_seconds").help("Duration of time user function ran in seconds.").register();
+
+        @Override
+        public void handle(HttpExchange he) throws IOException {
+            Histogram.Timer requestTimer = requestLatency.startTimer();
+            try {
+                requests.inc();
+                Class<?> c = Class.forName("io.kubeless."+className);
```

Comment:
Since these reflection methods are expensive, I wonder if you could call them and precache the `Method` instance in the `main` method, otherwise they will be called in every function call, affecting performance.